src/IROperator.h

```diff
@@ -1907,6 +1909,51 @@ inline Expr likely_if_innermost(Expr e) {
                                 {e}, Internal::Call::PureIntrinsic);
 }
 
+
+/** Cast an expression to the halide type corresponding to the C++
+ * type T clamping to the minimum and maximum values of the result
+ * type. */
+template <typename T>
+Expr saturating_cast(Expr e) {
+    return saturating_cast(type_of<T>(), e);
+}
+
+/** Cast an expression to a new type, clamping to the minimum and
+ * maximum values of the result type. */
+inline Expr saturating_cast(Type t, Expr e) {
+    // For float to float, guarantee infinities are always pinned to range.
```

Comment:
Why do we do this? It seems reasonable for (double) infinity -> (float) infinity when casting double -> float? That actually seems like the preferred behavior vs. converting infinity to merely a large value.